pkg/etcd/s3.go

```diff
@@ -0,0 +1,232 @@
+package etcd
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/sirupsen/logrus"
+)
+
+// s3 maintains state for S3 functionality.
+type s3 struct {
+	config *config.Control
+	client *minio.Client
+}
+
+// newS3 creates a new value of type s3 pointer with a
+// copy of the config.Control pointer and initializes
+// a new Minio client.
+func newS3(ctx context.Context, config *config.Control) (*s3, error) {
+	tr := http.DefaultTransport
+	if config.EtcdS3EndpointCA != "" {
+		trCA, err := setTransportCA(tr, config.EtcdS3EndpointCA, config.EtcdS3SkipSSLVerify)
```

Comment:
Why do we disable certificate validation when using a custom CA cert?

Reply:
It's set to false by default and configurable by the user.